lcls_tools/common/data_analysis/fitting/fitting_tool.py

```diff
@@ -0,0 +1,150 @@
+
+        #  for double gaussian
+        #  will make new helper functions for peaks and widths
+        amplitude2 = amplitude / 3
+        nu = mu / 2
+        rho = sigma / 2
+        initial_params["double_gaussian"] = [
+            amplitude,
+            mu,
+            sigma,
+            amplitude2,
+            nu,
+            rho,
+            offset,
+        ]
+
+        return initial_params
+
+    def get_fit(self, best_fit: bool = False) -> dict:
+        """Return fit parameters to data y such that y = method(x,parameters)"""
+        fits = {}
+
+        for key, val in self.initial_params.items():
+            method = getattr(self, key)
+            fit_params = curve_fit(method, self.x, self.y, val)[0]
+
+            y_fitted = method(self.x, *fit_params)
+            rmse = self.calculate_rms_deviation(self.y, y_fitted)
+            print(method)
+            print(rmse)
+            fits[key] = fit_params
+        print("returning only best fit: ", best_fit)
+        return fits
+
+    def check_skewness(self, outcomes, mu, sigma):
+        """Checks for skewness in dataset, neg if mean<median<mode, pos if opposite"""
+        mode = statistics.mode(outcomes)
+        pearsons_coeff = (mu - mode) / sigma
+        print(pearsons_coeff)
+        return pearsons_coeff
+
+    def check_kurtosis(self):
+        """greater kurtosis higher the peak"""
+        """how fast tails approaching zero, more outliers with higher kurtosis"""
+        """positive excess - tails approach zero slower"""
+        """negative excess - tails approach zero faster"""
+        # do later
+        return 0
+
+    def find_peaks(self):
+        pass
+
+    def find_widths(self):
+        pass
+
+    def find_runs(self):
+        pass
+
+    def find_moments(self):
+        """mean, sigma, skewness, kurtosis"""
+        pass
+
+    def truncate_distribution(x, lower_bound: float = None, upper_bound: float = None):
+        if lower_bound is None:
+            lower_bound = x.min()
+        if upper_bound is None:
+            upper_bound = x.max()
+        truncated_x = np.clip(x, lower_bound, upper_bound)
+        return truncated_x
+
+    def calculate_rms_deviation(self, x: np.array, fit_x: np.array):
+        rms_deviation = np.sqrt(np.power(sum(x - fit_x), 2) / len(x))
+        return rms_deviation
+
+    def calculate_unbiased_rms_deviated(x: np.array = None):
+        mean = np.mean(x)
+        rms_deviation = np.sqrt(np.power(sum(x - mean), 2) / len(x))
+        return rms_deviation
+
+    @staticmethod
+    def gaussian(x, amp, mu, sig, offset):
+        """Gaussian Function"""
+        """need a way to guess params if amp =/"""
+        return amp * np.exp(-np.power(x - mu, 2.0) / (2 * np.power(sig, 2.0))) + offset
+
+    @staticmethod
+    def gaussian_with_linear_background(x, amp, mu, sig, offset, slope):
+        return (
+            amp * np.exp(-np.power(x - mu, 2.0) / (2 * np.power(sig, 2.0)))
```

Comment:
repeated code?

Reply:
In the pydantic model version, I was going to make a model the gaussian functions. It would simply this. I can also just edit this code for the initial PR

Reply:
ok so this will get updated in the next PR?